### PR TITLE
Manage special bash options when no shell is defined in a workflow

### LINKED ISSUE #2448: Manage special bash options when no shell is defined in a workflow
### Bug report info

```plain text
act version:            0.2.66
GOOS:                   linux
GOARCH:                 amd64
NumCPU:                 4
Docker host:            DOCKER_HOST environment variable is not set
Sockets found:
        /var/run/docker.sock
Config files:           
        /home/derek/.config/act/actrc:
                -P ubuntu-latest=catthehacker/ubuntu:act-latest
                -P ubuntu-22.04=catthehacker/ubuntu:act-22.04
                -P ubuntu-20.04=catthehacker/ubuntu:act-20.04
                -P ubuntu-18.04=catthehacker/ubuntu:act-18.04
Build info:
        Go version:            go1.23.0
        Module path:           command-line-arguments
        Main version:          
        Main path:             
        Main checksum:         
        Build settings:
                -buildmode:           exe
                -compiler:            gc
                -ldflags:             -X main.version=0.2.66
                DefaultGODEBUG:       asynctimerchan=1,gotypesalias=0,httplaxcontentlength=1,httpmuxgo121=1,httpservecontentkeepheaders=1,tls10server=1,tls3des=1,tlskyber=0,tlsrsakex=1,tlsunsafeekm=1,winreadlinkvolume=0,winsymlink=0,x509keypairleaf=0,x509negativeserial=1
                CGO_ENABLED:          1
                CGO_CFLAGS:           
                CGO_CPPFLAGS:         
                CGO_CXXFLAGS:         
                CGO_LDFLAGS:          
                GOARCH:               amd64
                GOOS:                 linux
                GOAMD64:              v1
Docker Engine:
        Engine version:        27.2.1
        Engine runtime:        runc
        Cgroup version:        1
        Cgroup driver:         cgroupfs
        Storage driver:        overlay2
        Registry URI:          https://index.docker.io/v1/
        OS:                    Linux Mint 20.3
        OS type:               linux
        OS version:            20.3
        OS arch:               x86_64
        OS kernel:             5.4.0-193-generic
        OS CPU:                4
        OS memory:             15880 MB
        Security options:
                name=apparmor
                name=seccomp,profile=builtin
```

### Command used with act

```sh
act -v -P ubuntu-latest=localhost:5000/act-ubuntu-mvn:latest --remote-name github -j check-dependencies  workflowdispatch
```

### Describe issue

The job fails when executing this bash expression on a file that does not contain any dependency which is updatable:

`nb_updatable_dependencies=$(cat updatable.txt | grep . | grep -v '^No dependencies' | wc -l)`

It fails because bash is called with :

`-o pipefail`

This option is not used in github actions.

This issue was already fixed : https://github.com/nektos/act/issues/528
But the "-o pipefail" option came back in this PR: https://github.com/nektos/act/pull/575

### Link to GitHub repository

https://github.com/sebastien-perpignane/cardgame

### Workflow content

```yml
name: "Check dependencies"

on:
  schedule:
    - cron: '35 1 * * *'

  workflow_dispatch:

jobs:
  check-dependencies:
    uses: sebastien-perpignane/my-workflows/.github/workflows/check-dependencies.yml@main
    with:
      java-version: '21'
      distribution: 'temurin'
```

### Relevant log output

```sh
[check-dependencies/Check maven dependency updates/check-dependency-updates]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/2] user= workdir=
[check-dependencies/Check maven dependency updates/check-dependency-updates] [DEBUG] Exec command '[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/2]'
[check-dependencies/Check maven dependency updates/check-dependency-updates] [DEBUG] Working directory '/home/derek/workspace/cardgame'
| [INFO] Scanning for projects...
| [INFO] 
| [INFO] -------------------< sebastien.perpignane:cardgame >--------------------
| [INFO] Building cardgame 0.0.1-SNAPSHOT
| [INFO]   from pom.xml
| [INFO] --------------------------------[ jar ]---------------------------------
| [INFO] 
| [INFO] --- versions:2.17.1:display-dependency-updates (default-cli) @ cardgame ---
| [INFO] No dependencies in Dependencies have newer versions.
| [INFO] 
| [INFO] No dependencies in Plugin Dependencies have newer versions.
| [INFO] 
| [INFO] ------------------------------------------------------------------------
| [INFO] BUILD SUCCESS
| [INFO] ------------------------------------------------------------------------
| [INFO] Total time:  1.957 s
| [INFO] Finished at: 2024-09-10T20:07:03Z
| [INFO] ------------------------------------------------------------------------
[check-dependencies/Check maven dependency updates/check-dependency-updates]   ❌  Failure - Main Check dependencies
[check-dependencies/Check maven dependency updates/check-dependency-updates] exitcode '1': failure
```

### Additional information

Dockerfile to build the custom image I use (act-ubuntu-mvn image) -> https://gist.github.com/sebastien-perpignane/ade17b1922d35147d08cb08169a6a388

I submitted a PR as it is a very simple fix, even for someone who does not know Go, like me :)

### PR DESCRIPTION
**Pull request by sebastien-perpignane**
Fixes #2448

### CODE REVIEW

**Review by wolfogre — DISMISSED** [review]

**Comment by sebastien-perpignane** [comment]
> Thanks for finishing this looks good to me
> 
> _I also never wrote anything in go, until modding the existing nektos/act_

are we waiting for @wolfogre approval before merging ?

**Pull request merged**
